### PR TITLE
CDK: Autogenerate reference docs 

### LINKED ISSUE #2761: santa: docs
## Tell us about the problem you're trying to solve
Once Santa is ready for v1 release, create reference documentation

### PR DESCRIPTION
**Pull request by yevhenii-ldv**
## What
*closes #2761.*

## How
*Using SphinX library v4.1.0*

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
Expand the checklist which is relevant for this PR. 

<details><summary> <strong> Connector checklist </strong> </summary>
<p>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Secrets are annotated with `airbyte_secret` in the connector's spec
- [ ] Credentials added to Github CI if needed and not already present. [instructions for injecting secrets into CI](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#using-credentials-in-ci). 
- [ ] Unit & integration tests added as appropriate (and are passing)
    * Community members: please provide proof of this succeeding locally e.g: screenshot or copy-paste acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] `/test connector=connectors/<name>` command as documented [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-an-existing-connector) is passing. 
    * Community members can skip this, Airbyters will run this for you. 
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] `README.md`
    - [ ] `docs/SUMMARY.md` if it's a new connector
    - [ ] Created or updated reference docs in `docs/integrations/<source or destination>/<name>`.
    - [ ] Changelog in the appropriate page in `docs/integrations/...`. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md` contains a reference to the new connector
    - [ ] Build status added to [build page](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/builds.md)
- [ ] Build is successful
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] No major blockers
- [ ] PR merged into master branch
- [ ] Follow up tickets have been created
- [ ] Associated tickets have been closed & stakeholders notified
</p>
</details>

<details><summary> <strong> Connector Generator checklist </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

**Comment by sherifnada** [comment]
The next steps on this PR will be to upload it to readthedocs.io


@yevhenii-ldv feel free to continue it or hand that part off to someone else

**Comment by htrueman** [comment]
Generated docs example https://airbyte-cdk.readthedocs.io/en/latest/

File: airbyte-cdk/python/setup.py [hunk 1]
```diff
@@ -54,6 +54,24 @@
         "requests",
         "vcrpy",
         "Deprecated~=1.2",
+        "Jinja2~=3.0",
```
Comment (avida) [inline airbyte-cdk/python/setup.py:57]: those dependencies are not required for egular cdk, can we move it under extras_require "docs" section and install it only when generating docs

File: airbyte-cdk/python/reference_docs/make.bat [hunk 2]
```diff
@@ -0,0 +1,35 @@
+@ECHO OFF
```
Comment (avida) [inline airbyte-cdk/python/reference_docs/make.bat:1]: Why do we need this bat file? We already have makefile

Comment (htrueman) [reply airbyte-cdk/python/reference_docs/make.bat:1]: Both makefile and make.bat are only needed for the local development. So I left that if someone is going to use windows. I'm not sure if that's even the case, but that was my purpose to keep that file.

**Review by avida — APPROVED** [review]

**Review by Zirochkaa — APPROVED** [review]
Why do we need `airbyte-cdk/python/dist/airbyte_cdk-0.1.25-py3.7.egg`?

**Review by sherifnada — COMMENTED** [review]
does this get published when running `publish-cdk` ? if not, is it possible to do that? (and add breadcrumbs/instructions in the CDK README about how it's happening)

Also, this is a great start. But the current layout of the docs is not very easy to use/navigate. It's very nested and it's not clear where to find any concepts. Do you have any ideas about how to make it better?

File: airbyte-cdk/python/setup.py [hunk 3]
```diff
@@ -56,7 +56,29 @@
...
+            "pytz~=2021.1",
+            "snowballstemmer~=2.1",
+            "Sphinx~=4.2",
+            "sphinxcontrib-applehelp~=1.0",
+            "sphinxcontrib-devhelp~=1.0",
+            "sphinxcontrib-htmlhelp~=2.0",
+            "sphinxcontrib-jsmath~=1.0",
+            "sphinxcontrib-qthelp~=1.0",
+            "sphinx-rtd-theme~=1.0",
+            "sphinxcontrib-serializinghtml~=1.1",
```
Comment (keu) [inline airbyte-cdk/python/setup.py:79]: I believe you just put dependencies from Sphinx 4.2 here, if so, please remove it, you only need Sphinx 4.2 and all extra plugins

**Review by sherifnada — COMMENTED** [review]
Can you add a docs section about how to update/iterate on this documentation? What does a developer on the CDK need to know to quickly update the docs and publish them? How can they preview their changes without publishing them to readthedocs?

**Review by keu — APPROVED** [review]

**Review by sherifnada — APPROVED** [review]
a few small comments then free to merge

**Pull request merged**